src/core/inlines.js

```diff
@@ -138,7 +138,11 @@ function inlineVariableMatches(matched) {
   // remove "|" at the beginning and at the end, then split at an optional `:`
   const matches = matched.slice(1, -1).split(":", 2);
   const [varName, type] = matches.map(s => s.trim());
-  return hyperHTML`<var data-type="${type}">${varName}</var>`;
+  const element = hyperHTML`<var>${varName}</var>`;
+  if (type) {
+    element.dataset.type = type;
+  }
```

Comment:
This makes me sad :disappointed:

Reply:
A thought:

```js
optdata(html`<var>${varName}</var>`, { type });
// where optdata() assigns `dataset.type` only when the value is not nil
```

Reply:
But it doesn't really declare it's potentially nil and thus can be undefined... Everything is currently implicitly nullable.